Microsoft.Toolkit.Uwp.UI.Controls/Graph/SharePointFiles/SharePointFiles.cs

```diff
@@ -0,0 +1,558 @@
+        private Button _share;
+        private Button _download;
+        private Button _delete;
+        private HyperlinkButton _error;
+        private Button _cancel;
+        private Button _hasMore;
+        private Grid _thumbnail;
+        private Windows.UI.Xaml.Controls.Image _thumbnailImage;
+        private ScrollViewer _details;
+        private TextBlock _status;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SharePointFiles"/> class.
+        /// </summary>
+        public SharePointFiles()
+        {
+            DefaultStyleKey = typeof(SharePointFiles);
+        }
+
+        private static async void GraphAccessTokenPropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            SharePointFiles control = d as SharePointFiles;
+            control._graphClient = Common.GetAuthenticatedClient(control.GraphAccessToken);
+            if (!string.IsNullOrEmpty(control.DriveUrl))
+            {
+                await control.InitDrive(control.DriveUrl);
+            }
+        }
+
+        private static async void DriveUrlPropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            SharePointFiles control = d as SharePointFiles;
+            if (control._graphClient != null && !string.IsNullOrWhiteSpace(control.DriveUrl))
+            {
+                if (Uri.IsWellFormedUriString(control.DriveUrl, UriKind.Absolute))
+                {
+                    await control.InitDrive(control.DriveUrl);
+                }
+            }
+        }
+
+        private static void DetailPanePropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            SharePointFiles control = d as SharePointFiles;
+            if (control.IsDetailPaneVisible)
+            {
+                control.ShowDetailsPane();
+            }
+        }
+
+        /// <summary>
+        /// Called when applying the control template.
+        /// </summary>
+        protected override void OnApplyTemplate()
+        {
+            _list = GetTemplateChild("list") as ListView;
+            _back = GetTemplateChild("back") as Button;
+            _upload = GetTemplateChild("upload") as Button;
+            _share = GetTemplateChild("share") as Button;
+            _download = GetTemplateChild("download") as Button;
+            _delete = GetTemplateChild("delete") as Button;
+            _error = GetTemplateChild("error") as HyperlinkButton;
+            _cancel = GetTemplateChild("cancel") as Button;
+            _hasMore = GetTemplateChild("hasMore") as Button;
+            _thumbnail = GetTemplateChild("thumbnail") as Grid;
+            _thumbnailImage = GetTemplateChild("thumbnailImage") as Windows.UI.Xaml.Controls.Image;
+            _details = GetTemplateChild("details") as ScrollViewer;
+            _status = GetTemplateChild("status") as TextBlock;
+            _list.SelectionChanged += List_SelectionChanged;
```

Comment:
null checks.  It'd be nice of the clicks were commands in the template.

Reply:
done